Content.Server/Pinpointer/ProximityBeeperSystem.cs

```diff
@@ -73,7 +73,11 @@ public void UpdateBeep(EntityUid uid, ProximityBeeperComponent? component = null
         }
 
         if (closestDistance is not { } distance)
+        {
+            // If we don't reset the timer here walking into range will result in the beeper beeping frantically as it catches up.
+            component.NextBeepTime = _timing.CurTime + component.MaxBeepInterval;
             return;
+        }
 
```

Comment:
Rather than writing data every single tick (and potentially networking it whenever this gets predicted which would spam comp states) seems like it'd be better to just clamp the minimum nextbeeptime where if it's < curtime - beepinterval set it to curtime + beepinterval.